firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -3569,6 +3566,14 @@ extension BrowserViewController: TabManagerDelegate {
                 // The webView can go gray if it was zombified due to memory pressure.
                 // When this happens, the URL is nil, so try restoring the page upon selection.
                 selectedTab.reload()
+                // If the webView is loading, we hide the lock
+                // icon and wait for did finish to get the lasted secure content status
+            } else if webView.isLoading {
+                self.urlBar.locationView.hideTrackingProtectionButton()
+                // If not, we show the lock icon with the secure content status of the webView
```

Comment:
Overall it makes sense but hiding the lock until its done loading is a UX change which I don't know who should we ask. Although I am not sure if there is anything else we can do 🤷🏼‍♂️ 

cc @mattreaganmozilla any thoughts here?

Reply:
> hiding the lock until its done loading is a UX change

Yeah I wonder if this should ideally be raised with the UX team first to get their input 🤔; I can imagine some potential confusion from folks who might notice the lock icon missing entirely during the page load. (If I'm understanding the change correctly)